src/Simplify.cpp

```diff
@@ -42,6 +42,9 @@ bool is_simple_const(Expr e) {
     if (const Broadcast *b = e.as<Broadcast>()) {
         return is_simple_const(b->value);
     }
+    if (const Cast *c = e.as<Cast>()) {
```

Comment:
The whole point of is_simple_const is that it doesn't include casts. Otherwise it's equivalent to is_const.

In any case, we should no longer ever encounter casts of immediates since the introduction of immediates of arbitrary type.